shell/client/grainview.js

```diff
@@ -401,9 +410,9 @@ GrainView.prototype.sessionStatus = function () {
 GrainView.prototype.route = function () {
   this._dep.depend();
   if (this._token) {
-    return "/shared/" + this._token;
+    return "/shared/" + this._token + this._path;
```

Comment:
Oops, I just realized that as this is now, the query and fragment never get through at all, even on the first run.